### PR TITLE
docs(configuration): add new options for resolve

### LINKED ISSUE #3979: Document webpack change: update enhanced-resolve
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11478

---

- imports field support
- resolve.fallback option

recommend resolve.fallback for polyfills

**What kind of change does this PR introduce?**
deps, feature

**Did you add tests for your changes?**
in enhanced-resolve

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
- `imports` field in package.json is supported
  - new `resolve.importsFields` option to list the field names
- new `resolve.fallback` option: like alias but only when normal resolving fails

### PR DESCRIPTION
**Pull request by jamesgeorge007**
closes #3979

### CODE REVIEW

File: src/content/configuration/resolve.md [hunk 1]
```diff
@@ -488,6 +489,42 @@ module.exports = {
 };
 ```
 
+### `resolve.importsFields`
+
+`string[]`
+
+Field names from `package.json` which are used to provide the internal requests of a package (requests starting with `#` are considered as internal).
```
Comment (EugeneHlushko) [inline src/content/configuration/resolve.md:496]: Field names? What do you mean?

Comment (jamesgeorge007) [reply src/content/configuration/resolve.md:496]: https://github.com/webpack/webpack/blob/8295202bc86b45b791c28215703f7ce4e2eac370/declarations/WebpackOptions.d.ts#L1294-L1297

File: src/content/configuration/resolve.md [hunk 2]
```diff
@@ -488,6 +489,42 @@ module.exports = {
 };
 ```
 
+### `resolve.importsFields`
+
+`string[]`
```
Comment (EugeneHlushko) [inline src/content/configuration/resolve.md:494]: https://webpack.js.org/contribute/writers-guide/#configuration-defaults-and-types
```suggestion
`[string]`
```

File: src/content/configuration/resolve.md (hunk 1 above)
Comment (EugeneHlushko) [reply src/content/configuration/resolve.md]: Ok it actually meant the names of the fields in the package.json to look at, got it now

**Review by EugeneHlushko — APPROVED** [review]

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**